satpy/writers/cf_writer.py

```diff
@@ -616,6 +632,7 @@ def save_datasets(self, datasets, filename=None, groups=None, header_attrs=None,
             root.attrs['Conventions'] = CF_VERSION
 
         # Remove satpy-specific kwargs
+        to_netcdf_kwargs = copy.deepcopy(to_netcdf_kwargs)  # may contain dictionaries (encoding)
         satpy_kwargs = ['overlay', 'decorate', 'config_files']
```

Comment:
I know you didn't do this, but it is too bad we have to do this. I'm sure it will come to bite us in the future.